blocks/redirects-formatter/redirects-formatter.js

```diff
@@ -0,0 +1,173 @@
+import { getLibs } from '../../scripts/utils.js';
+
+export const SELECT_ALL_REGIONS = 'Select All Regions';
+export const DESELECT_ALL_REGIONS = 'De-select All Regions';
+export const NO_LOCALE_ERROR = 'No locales selected from list';
+const COPY_TO_CLIPBOARD = 'Copy to clipboard';
+
+async function createLocaleCheckboxes(prefixGroup) {
+  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+
+  return Object.keys(prefixGroup).map((key) => {
+    const { prefix } = prefixGroup[key];
+    const currLocale = prefix === '' ? 'en' : prefix;
+    if (currLocale === 'langstore') return;
+    const checkbox = createTag('input', { class: 'locale-checkbox', type: 'checkbox', id: `${currLocale}`, name: `${currLocale}` });
+    const label = createTag('label', { class: 'locale-label', for: `${currLocale}` });
+    label.innerText = `${currLocale}`;
+
+    // eslint-disable-next-line consistent-return
+    return createTag('div', { class: 'checkbox-wrapper' }, [checkbox, label]);
+  });
+}
+
+export function parseUrlString(input) {
+  const pairs = input.split('\n');
+
+  return pairs.reduce((rdx, pairString) => {
+    const pair = pairString.split('\t');
+    rdx.push(pair);
+    return rdx;
+  }, []);
+}
+
+function handleError(e, eSection) {
+  const errorElem = document.querySelector('.error');
+  setTimeout(() => {
+    errorElem.innerText = '';
+    eSection.classList.remove('error-border');
+  }, 2000);
+  errorElem.innerText = e;
+  eSection.classList.add('error-border');
+}
+
+export function generateRedirectList(urls, locales, handler) {
+  const inputSection = document.querySelector('.redirects-text-area');
+  const checkboxSection = document.querySelector('.checkbox-container');
+
+  return urls.reduce((rdx, urlPair) => {
+    if (!locales.length) handler(NO_LOCALE_ERROR, checkboxSection);
+
+    locales.forEach((locale) => {
+      let from;
+      let to;
+      try {
+        from = new URL(urlPair[0]);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        handler(e.message, inputSection);
+        return;
+      }
+      try {
+        to = new URL(urlPair[1]);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        handler(e.message, inputSection);
+        return;
+      }
+      const fromPath = from.pathname.split('.html')[0];
+      rdx.push([`/${locale}${fromPath}`, `${to.origin}/${locale}${to.pathname}`]);
+    });
+    return rdx;
+  }, []);
+}
+
+export function stringifyListForExcel(urls) {
+  return urls.reduce((rdx, url) => {
+    // eslint-disable-next-line no-param-reassign
+    rdx += `${url[0]}\t${url[1]}\n`;
+    return rdx;
+  }, '');
+}
```

Comment:
nit, this could be one line return without the eslint-disable
```js
export function stringifyListForExcel(urls) {
  return urls.reduce((rdx, url) => `${rdx}${url[0]}\t${url[1]}\n`, '');
}
```